api-search/src/main/groovy/org/cedar/onestop/api/search/service/ElasticsearchService.groovy

```diff
@@ -10,39 +10,85 @@ import org.elasticsearch.client.RestClient
+import org.springframework.http.HttpStatus
 import org.springframework.stereotype.Service
 
 @Slf4j
 @Service
 class ElasticsearchService {
 
-  @Value('${elasticsearch.index.prefix:}${elasticsearch.index.search.name}')
-  private String SEARCH_INDEX
+  @Value('${elasticsearch.index.search.collection.name}')
+  private String COLLECTION_SEARCH_INDEX
 
-  @Value('${elasticsearch.index.search.collectionType}')
-  private String COLLECTION_TYPE
+  @Value('${elasticsearch.index.search.granule.name}')
+  private String GRANULE_SEARCH_INDEX
 
-  @Value('${elasticsearch.index.search.granuleType}')
-  private String GRANULE_TYPE
+  @Value('${elasticsearch.index.search.flattenedGranule.name}')
+  private String FLATTENED_GRANULE_SEARCH_INDEX
+
+  @Value('${elasticsearch.index.universal-type}')
+  private String TYPE
+
+  @Value('${elasticsearch.index.prefix:}')
+  private String PREFIX
 
   private SearchRequestParserService searchRequestParserService
 
   private RestClient restClient
 
-
   @Autowired
   ElasticsearchService(SearchRequestParserService searchRequestParserService, RestClient restClient) {
     this.searchRequestParserService = searchRequestParserService
     this.restClient = restClient
   }
 
-  Map search(Map searchParams) {
-    def response = queryElasticsearch(searchParams)
-    return response
+  Map searchFlattenedGranules(Map searchParams) {
+    return queryElasticsearch(searchParams, PREFIX+FLATTENED_GRANULE_SEARCH_INDEX)
+  }
+
+  Map searchGranules(Map searchParams) {
+    return queryElasticsearch(searchParams, PREFIX+GRANULE_SEARCH_INDEX)
+  }
+
+  Map searchCollections(Map searchParams) {
+    return queryElasticsearch(searchParams, PREFIX+COLLECTION_SEARCH_INDEX)
+  }
+
+  Map getCollectionById(String id) {
+    def getCollection = getById(COLLECTION_SEARCH_INDEX, id)
+
+    if(getCollection.data) {
+      // get the total number of granules for this collection id
+      String granuleEndpoint = "/$PREFIX$GRANULE_SEARCH_INDEX/_search"
+      HttpEntity granuleRequest = new NStringEntity(JsonOutput.toJson([
+          query: [
+              term: [
+                  internalParentIdentifier: id
+              ]
+          ],
+          size : 0
+      ]), ContentType.APPLICATION_JSON)
+      def granuleResponse = restClient.performRequest("GET", granuleEndpoint, Collections.EMPTY_MAP, granuleRequest)
+      def totalGranulesForCollection = parseResponse(granuleResponse).hits.total
+
+      getCollection.meta = [
+          totalGranules: totalGranulesForCollection
+      ]
+    }
+
+    return getCollection
+  }
+
+  Map getGranuleById(String id) {
+    return getById(GRANULE_SEARCH_INDEX, id)
+  }
+
+  Map getFlattenedGranuleById(String id) {
+    return getById(FLATTENED_GRANULE_SEARCH_INDEX, id)
   }
 
   Map totalCounts() {
-    String collectionEndpoint = "/$SEARCH_INDEX/$COLLECTION_TYPE/_search"
+    String collectionEndpoint = "/$PREFIX$COLLECTION_SEARCH_INDEX/_search"
```

Comment:
We should probably use bracket notation in the variables here for clarity and readability